src/components/ModerationStatusSelect.tsx

```diff
@@ -0,0 +1,99 @@
+import type { IconComponent } from '@hypothesis/frontend-shared';
+import {
+  FilterIcon,
+  CautionIcon,
+  RestrictedIcon,
+  CheckAllIcon,
+  DottedCircleIcon,
+  Select,
+} from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import { Fragment } from 'preact';
+
+import type { ModerationStatus } from '../helpers';
+import { moderationStatusToLabel } from '../helpers';
+
+export type ModerationStatusSelectProps = {
+  alignListbox?: 'right' | 'left';
+  disabled?: boolean;
+} & (
+  | {
+      /** This select is used to filter a list of annotations by moderation status */
+      mode: 'filter';
+      selected?: ModerationStatus;
+      onChange: (status?: ModerationStatus) => void;
+    }
+  | {
+      /** This select is used to set the moderation status of a specific annotation */
+      mode: 'select';
+      selected: ModerationStatus;
+      onChange: (status: ModerationStatus) => void;
+    }
+);
+
+type Option = {
+  icon: IconComponent;
+  label: string;
+};
+
+const options = new Map<ModerationStatus, Option>([
+  [
+    'PENDING',
+    { label: moderationStatusToLabel.PENDING, icon: DottedCircleIcon },
+  ],
+  ['APPROVED', { label: moderationStatusToLabel.APPROVED, icon: CheckAllIcon }],
+  ['DENIED', { label: moderationStatusToLabel.DENIED, icon: RestrictedIcon }],
+  ['SPAM', { label: moderationStatusToLabel.SPAM, icon: CautionIcon }],
+]);
+
+export default function ModerationStatusSelect({
```

Comment:
This could use a JSDoc comment to explain briefly what the component is for, as it is now part of a library decoupled from actual usage in an application.

Reply:
Yeah, good point